cloudplatform/connectivity-oauth/src/test/java/com/sap/cloud/sdk/cloudplatform/connectivity/OAuth2ServiceBindingDestinationLoaderTest.java

```diff
@@ -361,54 +360,41 @@ void testProxiedDestination()
                     baseDestination,
-                    URI.create("proxyUrl"),
+                    proxyUrl,
                     tokenUrl,
                     credentials,
                     OnBehalfOf.TECHNICAL_USER_CURRENT_TENANT);
 
         assertThat(result.getUri()).isEqualTo(baseUrl);
-        assertThat(result.getHeaders(baseUrl))
-            .containsExactlyInAnyOrder(
-                new Header(HttpHeaders.AUTHORIZATION, "some-auth"),
-                expectedProxyHeader,
-                new Header("foo", "bar"),
-                new Header("sap-language", "en"));
-
-        verify(sut, times(1))
-            .createHeaderProvider(
-                eq(tokenUrl),
-                eq(credentials),
-                eq(OnBehalfOf.TECHNICAL_USER_CURRENT_TENANT),
-                eq(HttpHeaders.PROXY_AUTHORIZATION));
-    }
-
-    @Test
-    void testResilienceIsAdded()
-    {
-        final DefaultHttpDestination baseDestination = DefaultHttpDestination.builder(baseUrl).name("foo").build();
-
-        sut = new OAuth2ServiceBindingDestinationLoader();
+        assertThat(result)
+            .as("The new destination should have additional proxy properties.")
+            .isNotEqualTo(baseDestination);
+        assertThat(result.getProxyConfiguration()).contains(ProxyConfiguration.of(proxyUrl));
+        assertThat(((DefaultHttpDestination) result).getCustomHeaderProviders()).contains(headerProviderMock);
 
-        HttpDestination result =
-            sut.toDestination(baseUrl, tokenUrl, credentials, OnBehalfOf.TECHNICAL_USER_CURRENT_TENANT, TEST_SERVICE);
-
-        assertThat(result.get(OAuth2HeaderProvider.PROPERTY_OAUTH2_RESILIENCE_CONFIG)).isNotEmpty();
-        ResilienceConfiguration config =
-            (ResilienceConfiguration) result.get(OAuth2HeaderProvider.PROPERTY_OAUTH2_RESILIENCE_CONFIG).get();
-        assertThat(config.identifier()).startsWith(TEST_SERVICE.toString());
-
-        result =
+        final HttpDestination secondInvocationResult =
             sut
                 .toProxiedDestination(
                     baseDestination,
-                    baseUrl,
+                    proxyUrl,
                     tokenUrl,
                     credentials,
                     OnBehalfOf.TECHNICAL_USER_CURRENT_TENANT);
 
-        assertThat(result.get(OAuth2HeaderProvider.PROPERTY_OAUTH2_RESILIENCE_CONFIG)).isNotEmpty();
-        config = (ResilienceConfiguration) result.get(OAuth2HeaderProvider.PROPERTY_OAUTH2_RESILIENCE_CONFIG).get();
-        assertThat(config.identifier()).startsWith(baseDestination.get(DestinationProperty.NAME).get());
+        assertThat(secondInvocationResult)
+            .as("There should not be a cache in place for proxied destinations.")
+            .isNotSameAs(result);
+        assertThat(secondInvocationResult)
+            .as("The destination objects should be equal so that they use the same HTTP client.")
+            .isEqualTo(result)
+            .isNotSameAs(result);
```

Comment:
[minor]

Duplicated assertion.

```suggestion
            .isEqualTo(result);
```